joblib/test/test_parallel.py

```diff
@@ -440,6 +441,28 @@ def test_multiple_spawning():
                   [delayed(_reload_joblib)() for i in range(10)])
 
 
+class TestParallelBackend(SequentialBackend):
+    pass
+
+
+def test_invalid_backend():
+    assert_raises(ValueError, Parallel, backend='unit-testing')
+
+
+def test_overwrite_default_backend():
+    register_parallel_backend("default", TestParallelBackend)
+    assert_equal(VALID_BACKENDS["default"], TestParallelBackend)
+    register_parallel_backend("default", VALID_BACKENDS["multiprocessing"])
+
```

Comment:
Sorry, I was not clear. I thought of adding another test function for the new TestParallelBackend class and in the initial `test_overwrite_default_backend` add an assert to verify the 'default' backend has correctly been updated. Something like that:
```python
def test_overwrite_default_backend():
    register_parallel_backend("default", VALID_BACKENDS["multiprocessing"])
    assert_equal(VALID_BACKENDS["default"], VALID_BACKENDS["multiprocessing"])

def test_register_parallel_backend():
    register_parallel_backend("unit-testing", TestParallelBackend)
    assert_true("unit-testing" in VALID_BACKENDS)
    assert_equal(VALID_BACKENDS["unit-testing"], TestParallelBackend)

Maybe this is a bit overkill though.